pyflakes/checker.py

```diff
@@ -870,6 +895,33 @@ def ignore(self, node):
     # additional node types
     COMPREHENSION = KEYWORD = FORMATTEDVALUE = handleChildren
 
+    def DICT(self, node):
+        keys = node.keys
+        keys = [self.convert_to_value(key) for key in keys]
+        for key in set(keys):
+            if keys.count(key) > 1:
+                if isinstance(key, VariableKey):
+                    self.report(messages.DuplicateVariableDictionaryKey,
+                                node, key.name)
+                else:
+                    self.report(messages.DuplicateDictionaryKey, node, key)
+        self.handleChildren(node)
+
+    def convert_to_value(self, item):
+        if isinstance(item, ast.Str):
+            return item.s
+        elif isinstance(item, ast.Tuple):
+            return tuple(self.convert_to_value(i) for i in item.elts)
+        elif isinstance(item, ast.Num):
+            return item.n
+        elif isinstance(item, ast.Name):
```

Comment:
Just to make sure I am understanding correctly:
e.g.
```language=python
class Test(object):
    pass
a = Test()
b = Test()
a.something = 'yes'
{
  a.something: 1,
  a.something: 1,
}```
If your assumption is correct, are you expecting this to fail (not detect a duplicate) or fail (traceback)?
I'll give it a test assuming that you meant the above for the moment (good test to have anyway), and if I misunderstood just let me know and I'll deal with it either later today, or tomorrow evening.

Apologies if I'm misunderstanding something trivial- this it the first time I've done much with the AST.